src/labeler.ts

```diff
@@ -38,24 +47,38 @@ export async function run() {
       configPath
     );
 
-    const labels: string[] = [];
+    const labelsToAdd: string[] = [];
     const labelsToRemove: string[] = [];
+    const preexistingLabels = pullRequest.labels
+      .map((l) => l.name)
+      .filter((l): l is string => !!l); // just to get the type to be string[] instead of (string|undefined)[]
+
     for (const [label, globs] of labelGlobs.entries()) {
       core.debug(`processing ${label}`);
       if (checkGlobs(changedFiles, globs)) {
-        labels.push(label);
-      } else if (pullRequest.labels.find((l) => l.name === label)) {
+        labelsToAdd.push(label);
+      } else if (preexistingLabels.find((l) => l === label)) {
```

Comment:
Do we really need to pre-populate preexistingLabels with the names of labels? Accessing the object property is fast and chip operation, but allocating memory for another array is slow and expensive, it can even cause extra GC in the worst case.